### DISCUSSION TITLE
is new s3fs library is in beta phase ?

### DISCUSSION THREAD

**Discussion by ghost** [dask/s3fs#673] [Q&A]
Hi team,

I want to use s3fs in one of my python projects, just wanted to check with you all if the new version 2022.11.0 is stable/production or still in the beta phase.
if the latest version is in the beta phase, then what is the production/stable version?

**Comment by ianthomas23** [comment]
The whole project is classified as beta phase rather than any specific version. My personal view is that even though it is classified as beta I consider it production stable; quite a lot of work is ongoing and planned but this is unlikely to cause any problem to regular users of the library. According to https://pypistats.org/packages/s3fs the library is downloaded 69 million times a month from PyPI, so that equates to a large number of users in lots of different environments happily using it.

**Reply by martindurant** [reply]
Agree totally. "beta" here means that some new features may appear, and those can be experimental, but the core functionality is not expected to change. We continue to maintain the library and fix bugs as they appear. It is very typical for packages in the pydata world to remain "beta" forever, or never release the mythical v1.0 ([good example](https://github.com/numba/numba/releases)), which is also why many have moved to calendar versioning to break the link with "major" versus "minor" changes.

**Reply by ghost** [reply]
Thanks for replying back on my discussion it helped me a lot.

**Comment by martindurant** [comment]
On second reading, perhaps you were asking whether that version was a "release candidate"? Some projects do have those before official releases, but we do not.

**Reply by ghost** [reply]
on https://pypi.org/project/s3fs/ development phase is mentioned as beta, hence before using this library in the project I wanted to get clarification on whether in the future it will get support or not. Thanks for the replies and Thanks for creating and maintaining this wonderful library.

**Comment by mrocklin** [comment]
It looks like on PyPI the metadata says "beta" (presumably coming from
setup.py).  Maybe it's time to change that metadata.

**Reply by ghost** [reply]
yes on https://pypi.org/project/s3fs/ development phase was mentioned as beta hence I got confused whether I can use it or not for my project

**Reply by ghost** [reply]
Thanks for replying back on my discussion it helped me a lot.
